examples/analysis/geospatial/proximity_metric.py

```diff
@@ -0,0 +1,51 @@
+"""
+Proximity to raster or vector
+=============================
+
+This example demonstrates the calculation of proximity distances to a raster or vector using :func:`~geoutils.Raster.proximity`.
+"""
+# %%
+# We open an example raster, and a vector for which we select a single feature
+
+# sphinx_gallery_thumbnail_number = 2
+import geoutils as gu
+
+rast = gu.Raster(gu.examples.get_path("everest_landsat_b4"))
+vect = gu.Vector(gu.examples.get_path("everest_rgi_outlines"))
+vect = gu.Vector(vect.ds[vect.ds["RGIId"] == "RGI60-15.10055"])
+rast.crop(vect)
+
+# Plot the raster and vector
+import matplotlib.pyplot as plt
+
+ax = plt.gca()
+rast.show(ax=ax, cmap="Blues")
+vect.reproject(rast).ds.plot(ax=ax, fc="none", ec="k", lw=2)
+
+# %%
+# We use the raster as a reference to match for rasterizing the proximity distances with :func:`~geoutils.Vector.proximity`. See :ref:`core-match-ref` for more details.
+
+proximity = vect.proximity(rast)
+proximity.show(cmap="viridis")
+
+# %%
+# Proximity can also be computed to target pixels of a raster, or that of a mask
+
+# Get mask of pixels within 30 of 200 infrared
+import numpy as np
+
+mask_200 = np.abs(rast - 200) < 30
+mask_200.show()
```

Comment:
Add `interpolation="none"` to avoid the interpolated values between 0 and 1.
I don't know how easy it would be, but maybe we should try to have a set of matplotlibrc parameters with geoutils?

Reply:
I'll open an issue on this

Reply:
Added to #361